PreSignUpLambdaHook/index.mjs

```diff
@@ -1,35 +1,26 @@
-﻿import pg from "pg";
-import fs from "fs";
+﻿import pg from 'pg';
+import fs from 'fs/promises';
 
-export const handler = async (event, context, callback) => {
+const checkEmailAddress = async (emailAddress) => {
     const client = new pg.Client({
         ssl: {
-            rejectUnauthorized: false,
-            ca: fs.readFileSync("certs/eu-west-2-bundle.pem").toString(),
+            ca: await fs.readFile('certs/eu-west-2-bundle.pem', 'utf-8'),
         },
     });
-    await client.connect();
-    const res = await client.query("SELECT email_address FROM local_authorities;");
 
-    const allowedEmailAddresses = res.rows.map(row => row.email_address);
+    await client.connect();
+    const res = await client.query(
+        'SELECT "EmailAddress" FROM "Users" WHERE LOWER("EmailAddress")=LOWER($1)',
```

Comment:
Canonically, it's better to do this by converting both to uppercase, not lowercase! (of course it is _best_ to do something clever with languages, but I don't think we need to worry about that)

Reply:
Interesting - canonically in what sense?

I think we don't want to do anything clever with languages unless we're very sure of what we're doing. The behaviour of the local part of the email address (before the @) is entirely client-dependent (and theoretically _can_ be case-sensitive - although it never is), so we'd need to be sure of matching those rules to avoid any risk of letting people sign up with dodgy non-ASCII characters.

Email address formats are a nightmare 😱